src/sidebar/services/annotations.js

```diff
@@ -59,7 +61,65 @@ export default function annotationsService(
     searchClient.get({ uri: uris, group: groupId });
   }
 
+  /**
+   * Apply changes for the given `annotation` from its draft in the store (if
+   * any) and return a new object with those changes integrated.
+   */
+  function applyDraftChanges(annotation) {
+    const changes = {};
+    const draft = store.getDraft(annotation);
+
+    if (draft) {
+      changes.tags = draft.tags;
+      changes.text = draft.text;
+      changes.permissions = draft.isPrivate
+        ? privatePermissions(annotation.user)
+        : sharedPermissions(annotation.user, annotation.group);
+    }
+
+    // Integrate changes from draft into object to be persisted
+    return Object.assign({}, annotation, changes);
```

Comment:
This is fine, though I'd recommend using spread syntax `{ ...annotation, ...changes }` in new code as its more succinct.

Reply:
Roger and done.